javaassignment.js

```diff
@@ -0,0 +1,381 @@
+    
+}else if (b<a && b<c)
+{
+    return b;
+    
+}else{
+
+    return c;
+    
+}
+}
+function comparision (a,b,c){
+    console.log ('the maximum number is ' + max(a,b,c));
+    console.log('the mimimun number is ' + min(a,b,c));
+}
+
+comparision(5,6,7);
+/*PS C:\Users\bisrat\Desktop\source> node max.js
+the maximum number is 7
+the mimimun number is 5
+
+********************************************************************
+question 11 */
+function workinghours(hrs){
+    let sum=0;
+    for ( i=0; i<hrs.length;i++){
+        sum=sum+hrs[i];
+    }
+    console.log('the sum of the working hours is  ' + sum);
+
+}
+
+
+workinghours([4,5,7.5,8.5,9,6]);
+workinghours([6,5.5,3.5,8,10,9]);
+
+/*PS C:\Users\bisrat\Desktop\source> node max.js
+the sum of the working hours is  40
+the sum of the working hours is  42
+**************************************************************
+
+
+question 13(area of shape) */
+
+function areacalculator(shape,side){
+    let area;
+    if (shape== 'circle' && side.length==1){
+
+
+            area= 0.5*side[0]*side[0];
+            return area;
+
+        
+    }
+    else if (shape== 'rectangle' && side.length==2)
+    {
+        
+            area=side[0]*side[1];
+            return area;
+        }
+    
+    else if ( shape== 'triangle'  && side.length==2){
+       
+            area=0.5*side[0]*side[1];
+            return area;
+        
+    }else{
+        console.log('wrong shape');
+    }
+
+}
+
+console.log('area of the circle is with radius 4 is eaual to '+ areacalculator('circle',[4]));
+console.log('area of the rectangle is with height 8 and width 5 is eaual to '+ areacalculator('rectangle',[8,5]));
+console.log('area of the triangle is with height 8 and width 5 is eaual to '+ areacalculator('triangle',[8,5]));
+/*PS C:\Users\bisrat\Desktop\source> node max.js
+area of the circle is with radius 4 is eaual to 8
+area of the rectangle is with height 8 and width 5 is eaual to 40
+area of the triangle is with height 8 and width 5 is eaual to 20
+
+**********************************************************************
+
+question 14 */
+
+const stock=[{
+    Name : 'pencil',
+    price : 1.50,
+    quantity : 5000
+
+},
+{
+    Name : 'pen',
+    price : 2.0,
+    quantity : 3000
+
+},
+{
+    Name : 'exercisebook',
+    price : 20,
+    quantity : 0
+
+},
+{
+    Name : 'ruler',
+    price : 1.50,
+    quantity : 0
+
+},
+{
+    Name : 'pencillead',
+    price : 1.5,
+    quantity : 7000
+
+},
+{
+    Name : 'mislaneous',
+    price : 20,
+    quantity : 0
+
+}
+
+];
+
+function stockamount(){
+    let totalstock=0;
+    for(i=0;i<stock.length;i++){
+    totalstock +=(stock[i].quantity)};
+    return totalstock;
+}
+  console.log('total stock on hand is :-    ' + stockamount());
+
+/************************************************************
+
+question 15 */
+
+function stockonhand(){
+    for(i=0;i<stock.length;i++){
+        if (stock[i].quantity>0){
+            console.log(stock[i].Name);

```

Comment:
this function should return a list of products that are in stock, remove the console log and implement the function in such a way that it returns a new array of products containing only those items that are in stock